sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/ParametersCorrectOrderBase.cs

```diff
@@ -0,0 +1,169 @@
+            List<string> identifierNames, SemanticModel semanticModel)
+        {
+            for (var i = 0; i < argumentIdentifiers.Count; i++)
+            {
+                var argumentIdentifier = argumentIdentifiers[i];
+                var identifierName = argumentIdentifier.IdentifierName;
+                var parameter = argumentParameterMappings[argumentIdentifier.ArgumentSyntax];
+                var parameterName = parameter.Name;
+
+                if (string.IsNullOrEmpty(identifierName) ||
+                    !parameterNames.Contains(identifierName) ||
+                    !IdentifierWithSameNameAndTypeExists(parameter))
+                {
+                    continue;
+                }
+
+                if (argumentIdentifier is PositionalArgumentIdentifier positional &&
+                    (parameter.IsParams || identifierName == parameterName))
+                {
+                    continue;
+                }
+
+                if (argumentIdentifier is NamedArgumentIdentifier named &&
+                    (!identifierNames.Contains(named.DeclaredName) || named.DeclaredName == named.IdentifierName))
+                {
+                    continue;
+                }
+
+                return true;
+            }
+
+            return false;
+
+            bool IdentifierWithSameNameAndTypeExists(IParameterSymbol parameter) =>
+                argumentIdentifiers.Any(ia =>
+                    ia.IdentifierName == parameter.Name &&
+                    GetArgumentTypeSymbolInfo(ia.ArgumentSyntax, semanticModel).ConvertedType.DerivesOrImplements(parameter.Type));
+        }
+
+        private ArgumentIdentifier ConvertToArgumentIdentifier(TArgumentSyntax argument, int index)
+        {
+            var identifierName = GetArgumentIdentifier(argument)?.Text;
+            var nameColonIdentifier = GetNameColonArgumentIdentifier(argument);
+
+            if (nameColonIdentifier == null)
+            {
+                return new PositionalArgumentIdentifier(identifierName, argument, index);
+            }
+
+            return new NamedArgumentIdentifier(identifierName, argument, nameColonIdentifier.Value.Text);
+        }
+
+        private class ArgumentIdentifier
+        {
+            public ArgumentIdentifier(string identifierName, TArgumentSyntax argumentSyntax)
+            {
+                IdentifierName = identifierName;
+                ArgumentSyntax = argumentSyntax;
+            }
+
+            public string IdentifierName { get; }
+            public TArgumentSyntax ArgumentSyntax { get; }
+        }
+
+        private class PositionalArgumentIdentifier : ArgumentIdentifier
+        {
+            public PositionalArgumentIdentifier(string identifierName, TArgumentSyntax argumentSyntax, int position)
+                : base(identifierName, argumentSyntax)
+            {
+                Position = position;
+            }
+
+            public int Position { get; }
```

Comment:
this is not used. Why do you need this class?

Reply:
Removed.